### DISCUSSION TITLE
feat: Support/Integration for PandasAI

### DISCUSSION THREAD

**Discussion by apexx77** [open-webui/open-webui#6317] [General]
# Feature Request
## Support/Integration for [PandasAI](https://github.com/Sinaptik-AI/pandas-ai)

PandasAI is an open-source Python platform that simplifies data analysis by allowing users to ask questions in natural language. It bridges the gap between humans and data, making data exploration accessible to non-technical users and enhancing the efficiency of data analysis for technical users.

Here's a breakdown of PandasAI's key features:

* **Natural Language Interaction:** Users can pose questions about their data in plain English, eliminating the need for complex coding.
* **Versatile Deployment:** PandasAI can be integrated into Jupyter notebooks, Streamlit applications, or deployed as a REST API using frameworks like FastAPI or Flask.

* **Supported Data Sources:**  PandasAI can analyze data from various sources, including CSV files, SQL databases, pandas DataFrames, and more.

* **LLM-Powered:** PandasAI leverages the power of Large Language Models (LLMs) like GPT-3.5, GPT-4, Anthropic models, and Vertex AI to understand and respond to user queries.

* **Privacy-Conscious:** PandasAI prioritizes data privacy by randomizing and sampling data before sending it to the LLM, minimizing the risk of exposing sensitive information.

* **Open Source and Community-Driven:** As an open-source project, PandasAI benefits from contributions and continuous improvement from a dedicated community.

It would be beneficial if OpenWebUI has support for PandasAI

**Comment by v1-valux** [comment]
+1

Lately I was fiddling around with functions and pipelines and I was wondering if/how you could implement it in there..
Currently (outside open-webui) I'm using a few lines of code with within a streamlit app to retrieve, calculate and plot data from csv files.

There is not much code you need at minimum I guess..
```
df = pandas.read_csv(uploaded file)
api_base = 'http://ollama:11434/v1/'
model = 'llama3.1'

llm = LocalLLM(api_base=api_base, model=model)
sdf = SmartDataframe(df, config={'llm': llm})
x = sdf.chat(question)
```

I'd really love to use this in a function or similar.

But as PandasAI is "relying" on LMs like RAG does, it would be an even better solution to have a "sort of" Interface to data structures (like RAG uses documents) powered by PandasAI.
For a consistent workflow you could upload special files (.csv etc.) to your Knowledge Collections like done with PDFs.
Possibly instead of routing .CSV files to RAG models you then could use them as input for PandasAI?

Anyhow - this feature would make so much sense for (RAG-capable) LLM User Interfaces imho..

**Reply by MrDick-11** [reply]
there is one tool - https://openwebui.com/t/ajayrajc/chat_with_csv
but it's not working
Any news on it?

**Comment by Shauryadhaka** [comment]
any update on this?
